cmd/minikube/cmd/start.go

```diff
@@ -113,6 +115,8 @@ func runStart(cmd *cobra.Command, args []string) {
 	libMachineClient := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 	defer libMachineClient.Close()
 
+	setProxy()
```

Comment:
> we can add proxy to cluster.MachineConfig struct but to pass it till docker daemon we should have such option in the docker-machine side also

I am not after passing this through to docker-machine. I am trying to make things a bit more explicit. setProxy does all and I would like to split things up a bit. 

* First get the proxy settings from Viper and process/validate them
* Then apply them for example to MachineConfig, which also allows you to pass this options into StartHost or any other helper method. Really this are machine config options, right? Also it might make things more testable. Right now, you need to test proxy settings at the same time as command line options, because setProxy deals with both. By separating these two things (processing the falgs and doing something with them), you open up more/better ways of testing this